source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -330,6 +352,8 @@ pub trait EnvoyHttpFilter {
   ///
   /// Returns true if the operation is successful.
   fn set_response_trailer(&mut self, key: &str, value: &[u8]) -> bool;
+
+  fn send_response(&mut self, status_code: u32, headers: &HeaderMap, body: &str);
```

Comment:
rather than take a third party data structure at our API, which I think we don't want to do, can we simply accept a vector of &str being key and &[u8] being value?
```suggestion
  fn send_response(&mut self, status_code: u32, headers: headers: Vec<(&str, &[u8])>, body: &str);
```

then we can not only avoid the copy when possible from the user side as well as they have full control over conversion. Then we can also remove header_map_to_ffi as well as `pub struct Header {` here. (btw you could've been able to use abi::envoy_dynamic_module_type_http_header

Reply:
could you also make the body optional? i hope the mockall mock gen won't complain...